crates/ironrdp-rdpdr-native/src/nix/simple.rs

```diff
@@ -0,0 +1,765 @@
+            Ok(vec![SvcMessage::from(RdpdrPdu::ClientDriveQueryDirectoryResponse(
+                ClientDriveQueryDirectoryResponse {
+                    device_io_reply: DeviceIoResponse::new(req_inner.device_io_request, NtStatus::NO_SUCH_FILE),
+                    buffer: None,
+                },
+            ))])
+        }
+    }
+}
+fn make_create_drive_resp(
+    device_io_request: DeviceIoRequest,
+    create_disposation: CreateDisposition,
+    file_id: u32,
+) -> PduResult<Vec<SvcMessage>> {
+    let io_response = DeviceIoResponse::new(device_io_request, NtStatus::SUCCESS);
+    let information = match create_disposation {
+        CreateDisposition::FILE_CREATE
+        | CreateDisposition::FILE_SUPERSEDE
+        | CreateDisposition::FILE_OPEN
+        | CreateDisposition::FILE_OVERWRITE => Information::FILE_SUPERSEDED,
+        CreateDisposition::FILE_OPEN_IF => Information::FILE_OPENED,
+        CreateDisposition::FILE_OVERWRITE_IF => Information::FILE_OVERWRITTEN,
+        _ => Information::empty(),
+    };
+    let res = RdpdrPdu::DeviceCreateResponse(DeviceCreateResponse {
+        device_io_reply: io_response,
+        file_id,
+        information,
+    });
+    Ok(vec![SvcMessage::from(res)])
+}
+pub(crate) fn create_drive(
+    backend: &mut SimpleRdpdrBackend,
+    req_inner: DeviceCreateRequest,
+) -> PduResult<Vec<SvcMessage>> {
+    let file_id = backend.file_id;
+    backend.file_id += 1;
+    let mut path = String::from(backend.file_base.as_str());
+    path.push_str(&req_inner.path.replace('\\', "/"));
+    // first process directory
+    match std::fs::metadata(&path) {
+        Ok(meta) => {
+            if meta.is_dir() {
+                if req_inner.create_disposition == CreateDisposition::FILE_CREATE {
+                    warn!("create directory,but it exists");
+                    let io_response = DeviceIoResponse::new(req_inner.device_io_request, NtStatus::UNSUCCESSFUL);
+                    let res = RdpdrPdu::DeviceCreateResponse(DeviceCreateResponse {
+                        device_io_reply: io_response,
+                        file_id,
+                        information: Information::empty(),
+                    });
+                    return Ok(vec![SvcMessage::from(res)]);
+                }
+                if req_inner.create_options.bits() & CreateOptions::FILE_NON_DIRECTORY_FILE.bits() != 0 {
+                    warn!("create file,but it is directory");
+                    let io_response = DeviceIoResponse::new(req_inner.device_io_request, NtStatus::UNSUCCESSFUL);
+                    let res = RdpdrPdu::DeviceCreateResponse(DeviceCreateResponse {
+                        device_io_reply: io_response,
+                        file_id,
+                        information: Information::empty(),
+                    });
+                    return Ok(vec![SvcMessage::from(res)]);
+                }
+                //这个我们先不执行，我们看看能不能统一
+                //我们可以统一到文件打开的条件中去
```

Comment:
issue (non-blocking): Use English in the comments, because very few of IronRDP contributor can read Chinese 😆

Something like this? I’ll leave you adjust accordingly.
```suggestion
                // Let's not execute this for now, let's see if we can unify it.
                // This can be unified with the condition for opening the file.
```

Reply:
ok, i am sorry for forgetting to modify this comments